core/src/parquet/read/values.rs

```diff
@@ -443,49 +443,21 @@ macro_rules! make_int_variant_impl {
     ($ty: ident, $native_ty: ty, $type_size: expr) => {
         impl PlainDecoding for $ty {
             fn decode(src: &mut PlainDecoderInner, dst: &mut ParquetMutableVector, num: usize) {
-                let num_bytes = 4 * num; // Parquet stores Int8/Int16 using 4 bytes
-
                 let src_data = &src.data;
-                let mut src_offset = src.offset;
                 let dst_slice = dst.value_buffer.as_slice_mut();
                 let mut dst_offset = dst.num_values * $type_size;
-
-                let mut i = 0;
-                let mut in_ptr = &src_data[src_offset..] as *const [u8] as *const u8 as *const u32;
-
-                while num - i >= 32 {
```

Comment:
So this loop is effectively not needed?

Reply:
Yes, I combined the two loops into one. I explained more in https://github.com/apache/datafusion-comet/pull/549#issuecomment-2159574852